syntax = "proto3";

package email_verifier;

service EmailVerifier {
  rpc CheckEmail (EmailRequest) returns (EmailReply) {}
  rpc UpdateEmail (EmailRequest) returns (EmailReply) {}
  rpc DeleteEmail (EmailRequest) returns (DeleteReply) {}
}

message EmailRequest {
  string email = 1;
}

message EmailReply {
  bool is_valid = 1;
  string domain = 2;
  bool is_common_domain = 3;
}

message DeleteReply {
  string status = 1;
}